### DISCUSSION TITLE
useAsyncData requires a non null and non undefined value in 3.12, why when false is ok?

### DISCUSSION THREAD

**Discussion by SebbeJohansson** [nuxt/nuxt#27899] [Questions] [answered]
Hi! What is the reasoning behind the warning about returning a null or undefined value for useAsyncData? In my head it would be logical to return undefined on failed requests. Especially when false is okay.
We have a lot of useAsyncData that either returns nothing, or it does `return;` on incorrect/broken/missing returns.

**Comment by trc-mathieu** [comment, marked as answer]
Have you seen the explanation in the migration guide?
https://nuxt.com/docs/getting-started/upgrade#default-data-and-error-values-in-useasyncdata-and-usefetch

**Reply by SebbeJohansson** [reply]
Hi! I have not seen that. I assumed that was only for the nuxt4 release so never checked that page.
Still not sure I understand why they couldn't have done it so that a null value was fine, but I guess it could be for backwards compatibility.

We are going to look at throwing an error when we don't get anything back. Returning for example false in those cases, and true in the cases where we don't return anything at all isn't that clean.

**Reply by trc-mathieu** [reply]
I don't know if you saw but you can revert to the old behaviour also for now
```
experimental: {
    defaults: {
      useAsyncData: {
        value: 'null',
        errorValue: 'null'
      }
    }
  }
  ```

**Reply by SebbeJohansson** [reply]
Hmm okay. Not in a position to try it out for a while, but this changes the behavior to the old way and makes the warning go away?

**Reply by trc-mathieu** [reply]
exactly
